File: android/app/src/main/java/com/eos/smartworking/util/DateUtils.kt
package com.eos.smartworking.util

import java.time.DayOfWeek
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.time.temporal.TemporalAdjusters
import java.util.Locale

/** Utility per date — equivalente a teamWatcher.ts */
object DateUtils {

    private val dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
    private val weekRangeFormatter = DateTimeFormatter.ofPattern("d MMM", Locale.ITALIAN)

    fun getCurrentWeekStart(): String {
        val today = LocalDate.now()
        return today.with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY)).format(dateFormatter)
    }

    fun normalizeToMonday(dateStr: String): String {
        val date = LocalDate.parse(dateStr, dateFormatter)
        return date.with(TemporalAdjusters.previousOrSame(DayOfWeek.MONDAY)).format(dateFormatter)
    }

    fun formatLocalDate(date: LocalDate): String = date.format(dateFormatter)

    fun formatWeekRange(start: String): String {
        val d = LocalDate.parse(start, dateFormatter)
        val end = d.plusDays(4)
        return "${d.format(weekRangeFormatter)} – ${end.format(weekRangeFormatter)}"
    }

    fun parseDate(dateStr: String): LocalDate = LocalDate.parse(dateStr, dateFormatter)
}

File: ios/EosSmartWorking/Utilities/DateUtils.swift
import Foundation

// MARK: - Date utilities
enum DateUtils {
    
    private static let dateFormatter: DateFormatter = {
        let f = DateFormatter()
        f.dateFormat = "yyyy-MM-dd"
        f.locale = Locale(identifier: "en_US_POSIX")
        return f
    }()
    
    private static let weekRangeFormatter: DateFormatter = {
        let f = DateFormatter()
        f.dateFormat = "d MMM"
        f.locale = Locale(identifier: "it_IT")
        return f
    }()
    
    static func getCurrentWeekStart() -> String {
        let today = Date()
        let calendar = Calendar.current
        let monday = calendar.date(from: calendar.dateComponents([.yearForWeekOfYear, .weekOfYear], from: today))!
        return dateFormatter.string(from: monday)
    }
    
    static func normalizeToMonday(_ dateStr: String) -> String {
        guard let date = dateFormatter.date(from: dateStr) else { return dateStr }
        let calendar = Calendar.current
        let monday = calendar.date(from: calendar.dateComponents([.yearForWeekOfYear, .weekOfYear], from: date))!
        return dateFormatter.string(from: monday)
    }
    
    static func formatLocalDate(_ date: Date) -> String {
        dateFormatter.string(from: date)
    }
    
    static func formatWeekRange(_ start: String) -> String {
        guard let d = dateFormatter.date(from: start) else { return start }
        let end = Calendar.current.date(byAdding: .day, value: 4, to: d)!
        return "\(weekRangeFormatter.string(from: d)) – \(weekRangeFormatter.string(from: end))"
    }
    
    static func parseDate(_ dateStr: String) -> Date {
        dateFormatter.date(from: dateStr) ?? Date()
    }
}
